src/client/components/autocomplete2/list.tsx

```diff
@@ -57,55 +61,93 @@ export class AutocompleteList extends Component<
     this.setState({ items: newItems })
   }
 
+  onDragEnd = newItems => {
+    const { onDragEnd } = this.props
+    this.setState({ items: newItems })
+    onDragEnd && onDragEnd(newItems)
+  }
+
+  renderListItems = () => {
+    const { formatListItem, isDraggable } = this.props
+    const { items } = this.state
+
+    return items.map((item, i) => {
+      const title = item ? item.title || item.name : ""
+      return (
+        <ListItem key={i} isDraggable={isDraggable}>
+          {formatListItem ? (
+            formatListItem()
+          ) : (
+            <Serif size="4t" color={color("purple100")}>
+              {title}
+            </Serif>
+          )}
+          <button
+            className="remove-button"
+            onClick={() => this.onRemoveItem(i)}
+          />
+        </ListItem>
+      )
+    })
+  }
+
   render() {
-    const { formatListItem, label } = this.props
+    const { isDraggable, label } = this.props
     const { items } = this.state
 
     return (
-      <div>
+      <AutocompleteListContainer>
         {label && (
           <Sans size="3t" weight="medium">
             {label}
           </Sans>
         )}
+
         {items.length > 0 && (
           <Box mt={1}>
-            {items.map((item, i) => {
-              const title = item ? item.title || item.name : ""
-              return (
-                <ListItem key={i}>
-                  {formatListItem ? (
-                    formatListItem()
-                  ) : (
-                    <Serif size="4t" color={color("purple100")}>
-                      {title}
-                    </Serif>
-                  )}
-                  <button
-                    className="remove-button"
-                    onClick={() => this.onRemoveItem(i)}
-                  />
-                </ListItem>
-              )
-            })}
+            {isDraggable ? (
+              <DragDropList items={items} onDragEnd={this.onDragEnd}>
+                {this.renderListItems()}
+              </DragDropList>
+            ) : (
+              this.renderListItems()
+            )}
           </Box>
         )}
         <Autocomplete {...this.props} />
-      </div>
+      </AutocompleteListContainer>
     )
   }
 }
 
-export const ListItem = styled(Flex)`
+export const ListItem = styled(Flex).attrs<{ isDraggable?: boolean }>({})`
```

Comment:
you can do the following to define Props for styled-components now
```typescript
styled(Flex)<{ isDraggable?: boolean }>`
  ...
`
```

Reply:
Actually we can't do this yet, need to bump up some tooling in Positron first.

Reply:
Ah ok, probably need to upgrade `styled-components` first

Reply:
Yeah exactly, I'll make a separate PR for that.